### PR TITLE
Update macro to include optional count and fix link issue

### PR DESCRIPTION
**Pull request by Joffcom**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -2,44 +2,63 @@
...
+			f'<a href="https://n8n.io/workflows/" target="_blank">search all templates</a>'
+			f'</span></div>'
+		)
+
 	@env.macro	
 	def workflowDemo(workflow_endpoint):
 		r = requests.get(url = workflow_endpoint)
 		wf_data = r.json()
-		template_url = f'https://n8n.io/workflows/{wf_data["id"]}-{wf_data["name"].lower().replace(" ", "-").replace(":", "")}/'
+		template_url = f'https://n8n.io/workflows/{wf_data["id"]}-{re.sub("[^A-Za-z0-9-]","",wf_data["name"].replace(" ", "-")).lower()}/'
```
Comment (ayatnkw) [inline main.py:61]: @Joffcom I just found danother broken url ... 
(e.g. [AI Crew to Automate Fundamental Stock Analysis - Q&A Workflow](https://n8n.io/workflows/2183)). 
> expected url: 2183-ai-crew-to-automate-fundamental-stock-analysis-qanda-workflow
> actual url: 2183-ai-crew-to-automate-fundamental-stock-analysis---qa-workflow--

The n8n website seems to be using [slugify](https://www.npmjs.com/package/slugify) to generate the template url (?) where some special characters are converted into certain strings (& > and) and also hyphens `-` in title are removed. Is there a way to get the template url from the website without regenerating it from the title here?

Comment (Joffcom) [reply main.py:61]: That is a good spot @ayatnkw let me update the PR to take this into consideration.

**Review by ayatnkw — CHANGES_REQUESTED** [review]
Hi @Joffcom, sorry only got around to this now. I tested the URLs (ran a script to test urls on build process), and there are still 50+ invalid links. It seems that the slugify npm and Python packages behave slightly differently, which might be causing the issue.

Links are broken when:

Specific characters mapped [here](https://github.com/simov/slugify/blob/4860d7efd6734af8c716f26cd246aa5b1e503163/slugify.js#L15) are included in the title.
Special characters are included in the title. The npm package simply removes them, while the Python package replaces them with the default separator, which is a -.
It might be easier to simply insert the workflow ID and let it redirect to the actual URL, or, we could also rewrite the slugify npm package in Python. What do you think?

**Review by freakwriter — APPROVED** [review]
I tested with a few of the pages I knew had issues in docs currently, and they seem resolved in this branch. Also checked some that previously had issues which were resolved, and they still seem fixed. So this seems good to go to me.

**Comment by ihortom** [comment]
Hey @Joffcom , this update seems to have broken some pages, in particular docs for Merge node https://docs.n8n.io/integrations/builtin/core-nodes/n8n-nodes-base.merge/. The variable `no_template` is undefined causing the error below

```
Traceback (most recent call last):
  File "/opt/buildhome/python3.8/lib/python3.8/site-packages/mkdocs_macros/plugin.py", line 527, in render
    return md_template.render(**page_variables)
  File "/opt/buildhome/python3.8/lib/python3.8/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/opt/buildhome/python3.8/lib/python3.8/site-packages/jinja2/environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 175, in top-level template code
  File "/opt/build/repo/main.py", line 82, in workflowDemo
    if no_template:
NameError: name 'no_template' is not defined
```

**Comment by imchairmanm** [comment]
Just wanted to confirm that I'm seeing this as well.

**Pull request merged**